cedar-policy-core/src/entities/conformance.rs

```diff
@@ -176,6 +179,68 @@ pub fn typecheck_value_against_schematype(
     }
 }
 
+/// Check whether the given `RestrictedExpr` is a valid instance of `SchemaType`
+pub fn does_restricted_expr_implement_schematype(
+    expr: BorrowedRestrictedExpr<'_>,
+    expr_ty: &SchemaType,
+    expected_ty: &SchemaType,
+    extensions: &Extensions<'_>,
+) -> bool {
```

Comment:
having `expr_ty` as an input to this function is a little confusing, given that the comments say that we're checking `expr` itself against `expected_ty`

Reply:
We could compute it within the function but that's wasteful (and requires duplicate error-handling)

Reply:
Removed in latest revision.